server/src/shared/auth/implementations/mock-user-auth-handler.ts

```diff
@@ -1,41 +1,39 @@
 import { Result } from '../../core/result'
-import { JWT_SECRET } from '../../core/secret'
-import { UserMap } from '../../../modules/users/mappers/user-map'
-import { AuthToken, UserAuthHandler, UserAuthHandlerCreateResponse, UserAuthHandlerLoginResponse, UserAuthHandlerLoginSuccess } from '../user-auth-handler'
+import { UserAuthHandler, UserAuthHandlerCreateOptions, UserAuthHandlerCreateResponse, UserAuthHandlerLoginOptions, UserAuthHandlerLoginResponse, UserAuthHandlerLoginSuccess } from '../user-auth-handler'
 import { AppError } from '../../core/app-error'
-import express from 'express'
+import { AuthCodeString } from '../../../modules/users/domain/value-objects/auth-code'
 import { mocks } from '../../../test-utils'
-import { CreateUserDTO } from '../../../modules/users/application/use-cases/create-user/create-user-dto'
-import { sign } from 'jsonwebtoken'
+import { CreateUserDTOBody } from '../../../modules/users/application/use-cases/create-user/create-user-dto'
+import { UserMap } from '../../../modules/users/mappers/user-map'
 
-export const JWT_EXPIRY_TIME_IN_SECONDS =  604800; //1 week = 604800 seconds
 
 export const LOGIN_ERROR = 'login_error'
 export const CREATE_ERROR = 'create_error'
  
 //add implementation-specific auth functions here
 export class MockUserAuthHandler implements UserAuthHandler {
 
-  async login(req: express.Request, _res: express.Response): Promise<UserAuthHandlerLoginResponse> {
+  async login(options: UserAuthHandlerLoginOptions): Promise<UserAuthHandlerLoginResponse> {
     return new Promise((resolve) => {
-        if (req.body.status == LOGIN_ERROR) {
+        if (options.req.body.status == LOGIN_ERROR) {
```

Comment:
Ooh I guess I missed this in the previous PR, but the idea of triggering error cases in mocks with specific inputs is really cool!

In fact, the whole dependency injection format of these handlers is awesome

Reply:
Yeh that was motivated by dependency injection, definitely a cleaner way to organize it but this will do for the timebeing.